google/cloud/storage/tests/object_integration_test.cc

```diff
@@ -994,11 +995,37 @@ TEST_F(ObjectIntegrationTest, RewriteSimple) {
 
   // Rewrite object into a new object.
   auto object_name = MakeRandomObjectName();
-  ObjectRewriter rewriter(
-      client.raw_client(),
-      internal::RewriteObjectRequest(
-          bucket_name, source_name, bucket_name, object_name, "",
-          ObjectMetadata().set_content_type("plain/text")));
+  ObjectMetadata rewritten_meta = client.RewriteObjectBlocking(
+      bucket_name, source_name, bucket_name, object_name,
+      ObjectMetadata().set_content_type("plain/text"));
+
+  EXPECT_EQ(bucket_name, rewritten_meta.bucket());
+  EXPECT_EQ(object_name, rewritten_meta.name());
+
+  client.DeleteObject(bucket_name, object_name);
+  client.DeleteObject(bucket_name, source_name);
+}
+
+TEST_F(ObjectIntegrationTest, RewriteEncrypted) {
```

Comment:
At some point, we should write tests that test rewriting using CMEKs.  However, this is more difficult (to test against the actual service, anyway), since you have to specify a key that actually exists and that the person/project running the test can access/use.

Reply:
There is a CMEK key configured for the CI build that runs against production.  So in theory we can do this.